src/modules/campaign-monitor/services/campaign-monitor.service.spec.ts

```diff
@@ -386,6 +386,42 @@ describe('CampaignMonitorService', () => {
 				mockNewsletterTemplateDataWithNewsletter.CustomFields[7]
 			);
 		});
+
+		it('should parse preferences to newsletterTemplateData', () => {
+			const result = campaignMonitorService.convertPreferencesToNewsletterTemplateData(
+				mockUserInfo,
+				true
+			);
+			expect(result.EmailAddress).toEqual(
+				mockNewsletterTemplateDataWithNewsletter.EmailAddress
+			);
+			expect(result.Name).toEqual(mockNewsletterTemplateDataWithNewsletter.Name);
+			expect(result.Resubscribe).toEqual(
+				mockNewsletterTemplateDataWithNewsletter.Resubscribe
+			);
+			expect(result.ConsentToTrack).toEqual(
+				mockNewsletterTemplateDataWithNewsletter.ConsentToTrack
+			);
+			expect(result.CustomFields[0]).toEqual(
+				mockNewsletterTemplateDataWithNewsletter.CustomFields[0]
+			);
+			expect(result.CustomFields[1]).toEqual(
+				mockNewsletterTemplateDataWithNewsletter.CustomFields[1]
+			);
+			expect(result.CustomFields[2]).toEqual(
+				mockNewsletterTemplateDataWithNewsletter.CustomFields[2]
+			);
+			expect(result.CustomFields[3]).toEqual(
+				mockNewsletterTemplateDataWithNewsletter.CustomFields[3]
+			);
+			expect(result.CustomFields[4]).toEqual(
+				mockNewsletterTemplateDataWithNewsletter.CustomFields[4]
+			);
+			expect(result.CustomFields[6]).toEqual(
+				mockNewsletterTemplateDataWithNewsletter.CustomFields[6]
+			);
```

Comment:
did you forget index 5, or is it not needed to be checked?
In that case add a comment to the code. eg:
```
// field [5] is .... and we don't care about that this value is because ....
```